### ISSUE TITLE
Add option to consider variables magic numbers for no-magic-numbers

### ISSUE THREAD

**Issue by LJNeon** [eslint/eslint#10752]
**What rule do you want to change?**
`no-magic-numbers`
**Does this change cause the rule to produce more or fewer warnings?**
`more`
**How will the change be implemented? (New option, new default behavior, etc.)?**
`new option`
**Please provide some example code that this change will affect:**
```js
const a = 5;
let b = -13;
var c = 9;
```
**What does the rule currently do for this code?**
Doesn't treat the numbers as magic numbers.
**What will the rule do after it's changed?**
Treat all of the above as magic numbers when a boolean option is set. We need this because we want all of our constants to be in a single YAML file, not messily scattered around our code.

**Comment by kaicataldo** [comment]
It seems like this option would essentially disallow what the rule is trying to currently enforce.

> We need this because we want all of our constants to be in a single YAML file, not messily scattered around our code.

This seems like a really specific use-case to me, which makes me think that it might be better as a custom rule. Using AST selectors, it should be as simple as:
```
module.exports = function(context) {
  return {
    'VariableDeclarator[init.type="Literal"]'(node) {
      
      if (typeof node.init.value === "number") {
        context.report(node, 'No magic numbers. Instead, please put this in the YAML file.');
      }
    }
  };
};
```

I whipped that up really quickly - you'll have to double check it - but please let us know if that solves your use case.
